### PR TITLE
✨ Consultancy page about section second#180

### LINKED ISSUE #180: 🎨 Create Consultancy Page About Section (second)
#  Create Consultancy Page About Section (second)
---

#### Note: This issue is dependent on #8, please complete that before you continue.

Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a user, I want a component that displays the second section of the about section on the consultancy page. I want a good design on both desktop and mobile.

### Description

Your assignment is to implement the component to appear as below.  This component reuses the banner component created in #8. Please take note of all the design features.

consultancy-second-info

#### Link to full design:
```
https://xd.adobe.com/view/b98caf27-6549-49e5-8ed8-c5234788df61-ecdf/
```
### Technical Analysis

This component will be created in `libs\features\pages\consultancy-page` inside a components folder. The component will be called elewa-consultancy-about-two. The component reuses the image and text banner and is therefore dependent on #8. If you need to edit the code in #8 to get the desired result feel free to do so.

The section has a property of the below type:

```typeScript
 ImageAndText
```

NB: The interface for the type above will be implemented in #8

### Data
The mock data for this should be stored in `libs\data\sections`

### Acceptance Criteria

1. When rendered the component is pixel-perfect compared to the design above.
2. The component is clean and does not have more than 50 lines
3. The component SCSS uses CSS variables for colour, font, ... Connected to the main style.scss
6. The design is responsive for screen, tablet and phone
7. The image renders fast and has multiple srcsets

### PR DESCRIPTION
**Pull request by Marionwanyama23**
# Description

This pull request contains a component that displays the second section of the about section on the consultancy page. It is responsive to both mobile and desktop

Fixes #180 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshot 
Screenshot from 2023-09-11 15-38-52

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/data/sections/src/lib/consultancy-about-two.data.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  
+];
+
+export function getLayoutClass(item: ImageAndText): string {
+  
+  if (item && item.imagePosition) {
+    return item.imagePosition === 'right' ? 'right-image' : 'left-image';
+  }
+  return 'default-layout';
+}
```
Comment (LemmyMwaura) [inline libs/data/sections/src/lib/consultancy-about-two.data.ts:31]: Why is this method here, just have it in your component and remove the unnecessary if statement. The imagePosition is a required property so the if check is redundant.

File: libs/features/pages/consultancy-page/src/lib/components/elewa-consultancy-about-two/elewa-consultancy-about-two.component.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+})
+export class ElewaConsultancyAboutTwoComponent implements OnInit {
+  @Input() imageContent !: ImageAndText[];
+  
+  ngOnInit(){
+    this.imageContent = highlightedImageData;
+  }
+
+  getLayoutClass(content: ImageAndText): string {
+    return getLayoutClass(content);
```
Comment (LemmyMwaura) [inline libs/features/pages/consultancy-page/src/lib/components/elewa-consultancy-about-two/elewa-consultancy-about-two.component.ts:19]: The logic to get the class should be here not the in the mock data

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Changes left

File: libs/features/pages/consultancy-page/src/lib/components/elewa-consultancy-about-two/elewa-consultancy-about-two.component.ts (hunk 2 above)
Comment (Marionwanyama23) [reply libs/features/pages/consultancy-page/src/lib/components/elewa-consultancy-about-two/elewa-consultancy-about-two.component.ts]: I have made the necessary changes

**Review by LemmyMwaura — COMMENTED** [review]
Good job on this everything was clean apart from how data is being passed. Should be a property in the about component rather than an input (also wrong for the about one component), you can pull the latest changes just to see how it's been done.

**Comment by LemmyMwaura** [comment]
Closing this as it's been resolved

**Pull request closed without merging**